examples/oidc-sso/keycloak_oidc.md

```diff
@@ -0,0 +1,134 @@
+      scope: 'openid profile roles role_list email' <-- Declared in a realm Client Scopes
+      usernameParameter: 'preferred_username'
+      groupsParameter: 'groups'
+      kibanaExternalHost: 'localhost:5601'
+      logoutUrl: 'http://localhost:8080/auth/realms/ror/protocol/openid-connect/logout' <-- Value from from OpenID Endpoint Configuration
+      jwksURL: 'http://localhost:8080/auth/realms/ror/protocol/openid-connect/certs' <-- Value from from OpenID Endpoint Configuration
+```
+
+
+To verify all OpenID Endpoint Configuration-based, you can open OpenID Endpoint Configuration page in the kibana realm
+
+![keycloak\_screenshot](<../../.gitbook/assets/oidc_openid_endpoint_configuration.png>)
+
+To provide clientSecret value, you need to open ror_oidc client (or your custom client name)
+
+![keycloak\_screenshot](<../../.gitbook/assets/oidc_client_secret.png>)
+
+### Setup Elasticsearch with ReadonlyREST
+
+Our elasticsearch can be run with or without SSL. To make it available on HTTPS (more detailed info in our [documentation](https://github.com/beshu-tech/readonlyrest-docs/blob/master/elasticsearch.md#encryption)), so we modify the elasticsearch.yml
+
+**append to elasticsearch.yml**
+
+```
+xpack.security.enabled: false
+http.type: ssl_netty4 # <-- needed for ROR SSL
+```
+
+Write in **readonlyrest.yml**
+
+```
+readonlyrest:
+
+    ssl:
+      enable: true
+      keystore_file: "keystore.jks"
+      keystore_pass: readonlyrest
+      key_pass: readonlyrest
+
+    prompt_for_basic_auth: false
+
+    audit_collector: true
+
+    access_control_rules:
+    - name: "::KIBANA-SRV::"
+      auth_key: kibana:kibana
+      verbosity: error
+
+    - name: "ReadonlyREST Enterprise instance #1"
+      kibana_index: ".kibana_sso"
+      ror_kbn_auth:
+        name: "kbn1"
+
+    ror_kbn:
+    - name: kbn1
+      signature_key: "9yzBfnLaTYLfGPzyKW9es76RKYhUVgmuv6ZtehaScj5msGpBpa5FWpwk295uJYaaffTFnQC5tsknh2AguVDaTrqCLfM5zCTqdE4UGNL73h28Bg4dPrvTAFQyygQqv4xfgnevBED6VZYdfjXAQLc8J8ywaHQQSmprZqYCWGE6sM3vzNUEWWB3kmGrEKa4sGbXhmXZCvL6NDnEJhXPDJAzu9BMQxn8CzVLqrx6BxDgPYF8gZCxtyxMckXwCaYXrxAGbjkYH69F4wYhuAdHSWgRAQCuWwYmWCA6g39j4VPge5pv962XYvxwJpvn23Y5KvNZ5S5c6crdG4f4gTCXnU36x92fKMQzsQV9K4phcuNvMWkpqVB6xMA5aPzUeHcGytD93dG8D52P5BxsgaJJE6QqDrk3Y2vyLw9ZEbJhPRJxbuBKVCBtVx26Ldd46dq5eyyzmNEyQGLrjQ4qd978VtG8TNT5rkn4ETJQEju5HfCBbjm3urGLFVqxhGVawecT4YM9Rry4EqXWkRJGTFQWQRnweUFbKNbVTC9NxcXEp6K5rSPEy9trb5UYLYhhMJ9fWSBMuenGRjNSJxeurMRCaxPpNppBLFnp8qW5ezfHgCBpEjkSNNzP4uXMZFAXmdUfJ8XQdPTWuYfdHYc5TZWnzrdq9wcfFQRDpDB2zX5Myu96krDt9vA7wNKfYwkSczA6qUQV66jA8nV4Cs38cDAKVBXnxz22ddAVrPv8ajpu7hgBtULMURjvLt94Nc5FDKw79CTTQxffWEj9BJCDCpQnTufmT8xenywwVJvtj49yv2MP2mGECrVDRmcGUAYBKR8G6ZnFAYDVC9UhY46FGWDcyVX3HKwgtHeb45Ww7dsW8JdMnZYctaEU585GZmqTJp2LcAWRcQPH25JewnPX8pjzVpJNcy7avfA2bcU86bfASvQBDUCrhjgRmK2ECR6vzPwTsYKRgFrDqb62FeMdrKgJ9vKs435T5ACN7MNtdRXHQ4fj5pNpUMDW26Wd7tt9bkBTqEGf"
```

Comment:
Maybe let's add a comment here saying it has to be the same string as we had in kibana.yml

Reply:
Good idea! updated